cmd/hidi/main.go

```diff
@@ -0,0 +1,531 @@
+	}
+	return data
+}
+
+func newLogBuffer(size int) logBuffer {
+	return logBuffer{
+		buffer:   make([][]byte, size),
+		size:     size,
+		position: 0,
+	}
+}
+
+func runBinary(wg *sync.WaitGroup, ctx context.Context, port int) {
+	defer wg.Done()
+	exe, err := memexec.New(OpenRGB)
+	if err != nil {
+		panic(err)
+	}
+	wg.Add(1)
+	defer func() {
+		defer wg.Done()
+		err := exe.Close()
+		if err != nil {
+			log.Info(fmt.Sprintf("failed to close memory exec: %s", err), logger.Error)
+		}
+	}()
+
+	cmd := exe.Command("--server", "--noautoconnect", "--server-port", fmt.Sprintf("%d", port))
+
+	out1, in1 := io.Pipe()
+	out2, in2 := io.Pipe()
+
+	defer out1.Close()
+	defer in1.Close()
+	defer out2.Close()
+	defer in2.Close()
+
+	cmd.Stdout = in1
+	cmd.Stderr = in2
+
+	log.Info("[OpenRGB] start", logger.Debug)
+	err = cmd.Start()
+	if err != nil {
+		log.Info(fmt.Sprintf("[OpenRGB] Failed to start: %s", err), logger.Error)
+		return
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		<-ctx.Done()
+		time.Sleep(time.Millisecond * 200)
+		err := cmd.Process.Signal(os.Interrupt)
+		if err != nil {
+			log.Info(fmt.Sprintf("[OpenRGB] failed to send signal: %s", err), logger.Error)
+		} else {
+			log.Info("[OpenRGB] interrupt success", logger.Info)
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		scan1 := bufio.NewScanner(out1)
+		for scan1.Scan() {
+			log.Info("[OpenRGB] o> "+scan1.Text(), logger.Debug)
+		}
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		scan2 := bufio.NewScanner(out2)
+		for scan2.Scan() {
+			log.Info("[OpenRGB] e> "+scan2.Text(), logger.Debug)
+		}
+	}()
+
+	err = cmd.Wait()
+	if err != nil {
+		log.Info(fmt.Sprintf("[OpenRGB] Execution error: %s", err), logger.Error)
+	}
+	log.Info("[OpenRGB] Done", logger.Debug)
+
+}
+
+func main() {
+	if *force256 == true {
+		os.Setenv("TERM", "xterm-256color")
+	}
+	err := createConfigDirectoryIfNeeded()
+	if err != nil {
+		log.Info(fmt.Sprintf("configuration upkeep task failed: %s", err), logger.Warning)
+	}
+
+	var cfg = LoadHIDIConfig(configDir + "/hidi.config")
+	log.Info(fmt.Sprintf("HIDI config: %+v", cfg), logger.Debug)
+
+	var sigs = make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	g := runUI(cfg, *ui && !*silent, sigs)
+
+	// this wait-group has to be propagated everywhere where usual logging appear
+	wg := sync.WaitGroup{}
+
+	port := rand.Intn(65535 - 1024)
+	port += 1024
+
+	if *orgb {
+		if len(OpenRGB) != 0 {
+			log.Info(fmt.Sprintf("starting OpenRGB server (%s)", OpenRGBVersion), logger.Info)
+			wg.Add(1)
+			go runBinary(&wg, ctx, port)
+		} else {
+			log.Info("OpenRGB is not supported in that build", logger.Warning)
+		}
+	}
+
+	server := runProfileServer(&wg)
+
+	wg.Add(1)
+	go handleSigs(&wg, sigs, cancel, server, g)
+
+	ioDevices := midi.DetectDevices()
+	if len(ioDevices) == 0 {
+		// TODO: able to log things
```

Comment:
I think this one could be resolved, also it's critical to provide an information when something is preventing application from running  
`fmt.Print` should be good enough